lib.js

```diff
@@ -42,30 +45,53 @@ module.exports = ({
   it is recommended to customize the prefix.`);
   }
 
-  return Promise.all(templates.filter((template) => {
-    if (template.platform) {
-      return (platforms.indexOf(template.platform) >= 0);
-    }
-
-    return true;
-  }).map((template) => {
-    if (!template.name) {
-      return Promise.resolve();
-    }
-
-    const args = {
-      name: `${prefix}${pascalCase(name)}`,
-      moduleName: `${modulePrefix}-${paramCase(name)}`,
-      packageIdentifier,
-      namespace: namespace || pascalCase(name).split(/(?=[A-Z])/).join('.'),
-      platforms,
-    };
+  return Promise
+    .resolve()
+    .then(() => {
+      if (!generateExample) {
+        return
+      }
+      // Note: The example has to be created first because it will fail if there
+      // is already a package.json in the folder in which the command is executed.
+      return createFolder(name).then(() => {
+        execSync('react-native init example', { cwd: './' + name, stdio:'inherit'});
+      });
+    })
+    .then(() => {
+      return Promise.all(templates.filter((template) => {
+        if (template.platform) {
+          return (platforms.indexOf(template.platform) >= 0);
+        }
 
-    const filename = path.join(name, template.name(args));
-    const baseDir = filename.split(path.basename(filename))[0];
+        return true;
+      }).map((template) => {
+        if (!template.name) {
+          return Promise.resolve();
+        }
+        const args = {
+          name: `${prefix}${pascalCase(name)}`,
+          moduleName: `${modulePrefix}-${paramCase(name)}`,
+          packageIdentifier,
+          namespace: namespace || pascalCase(name).split(/(?=[A-Z])/).join('.'),
+          platforms,
+        };
 
-    return createFolder(baseDir).then(() =>
-      createFile(filename, template.content(args))
-    );
-  }));
+        const filename = path.join(name, template.name(args));
+        var baseDir = filename.split(path.basename(filename))[0];
+        
+        return createFolder(baseDir).then(() =>
+          createFile(filename, template.content(args))
+        );
+      }));
+    })
+    .then(() => {
+      if (!generateExample) {
+        return;
+      }
+      // Adds and links the created library project
+      const pathExampleApp = './' + name + '/example';
```

Comment:
Nitpick: What do you think about using template literals here? (It's just a personal preference on my part.)

Reply:
Sure. Will look at updating this.

Reply:
updated to use template literals.